packages/core/src/core/coreToolScheduler.test.ts

```diff
@@ -404,8 +405,30 @@ describe('CoreToolScheduler', () => {
 
 describe('CoreToolScheduler with payload', () => {
   it('should update args and diff and execute tool when payload is provided', async () => {
-    const mockTool = new MockModifiableTool();
-    mockTool.executeFn = vi.fn();
+    const mockingParams: MockModifiableToolShouldConfirmExecuteParams = {
+      filePath: 'test.txt',
+      currentContent: 'old content',
+      proposedContent: 'new content',
+    };
+
+    const mockCreateUpdatesParamsFn = (
+      _oldContent: string,
+      modifiedProposedContent: string,
+      _originalParams: Record<string, unknown>,
+    ): Record<string, unknown> => ({
+      newContent: modifiedProposedContent,
+    });
+
+    const executeFn = vi.fn().mockResolvedValue({
```

Comment:
Super nit: `vi.fn()` isn't very typesafe by itself. Consider adding a type to this

vi.fn<typeof SomeFunction>().mockResolvedValue();